### PR TITLE
Add Indic STS benchmark dataset

### PR DESCRIPTION
**Pull request by jaygala24**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [ ] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

**Comment by jaygala24** [comment]
Add Semantic Textual Similarity benchmark between English and 12 high-resource Indic languages.

Team:
- Diganta Mista (@digantamisra98) - MILA
- Jay Gala (@jaygala24) - AI4Bharat

File: mteb/tasks/STS/multilingual/IndicCrosslingualSTS.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    volume = {10},
+    pages = {145-162},
+    year = {2022},
+    month = {02},
+    issn = {2307-387X},
+    doi = {10.1162/tacl_a_00452},
+    url = {https://doi.org/10.1162/tacl\_a\_00452},
+    eprint = {https://direct.mit.edu/tacl/article-pdf/doi/10.1162/tacl\_a\_00452/1987010/tacl\_a\_00452.pdf},
+}""",
+        n_samples={"test": 10020},
```
Comment (KennethEnevoldsen) [inline mteb/tasks/STS/multilingual/IndicCrosslingualSTS.py:61]: This becomes quite a large dataset. It might be worth reducing the size of the splits to 256 pr. language (assuming it does not influence the scores too much, which I don't believe it will)

**Review by KennethEnevoldsen — COMMENTED** [review]
Looks great! Feel free to add the points as well. I only have one pointer related to size

File: mteb/tasks/STS/multilingual/IndicCrosslingualSTS.py (hunk 1 above)
Comment (jaygala24) [reply mteb/tasks/STS/multilingual/IndicCrosslingualSTS.py:61]: Yes, I acknowledge that the split size is higher but it doesn't take much time in evaluation. Both the models mentioned above took nearly 9 secs for evaluations. I will evaluate the models with the split size of 256 per language and check if the results are consistent with the larger splits.

Comment (jaygala24) [reply mteb/tasks/STS/multilingual/IndicCrosslingualSTS.py]: The split size of 10000 is across 12 Indic languages. The split size for each language is roughly around 800 which is well within the mentioned limit of 2048. I have updated the metadata for the dataset in the latest commit.

**Comment by jaygala24** [comment]
**Points Summary:**

This is the first dataset for Indic languages in the STS task category. Thus, all the 12 Indic languages are new ones for this task category.

- This dataset = 2 pts
- Any new language not covered previously in STS task = 4 pts x 12 languages

Total points = 2 + 4 x 12 = 50 pts

File: mteb/tasks/STS/multilingual/IndicCrosslingualSTS.py (hunk 1 above)
Comment (KennethEnevoldsen) [reply mteb/tasks/STS/multilingual/IndicCrosslingualSTS.py]: 2048 is a suggested limit no hard rules there. We are trying to keep MMTEB as small as possible (but naturally as big as required).

It is quite easy to experiment with downsampling in the `dataset_transform()`, then it will be easy enough to empirically estimate whether downsampling notably influences performance (STS tasks are quite robust to downsampling as far as I have seen).

Comment (jaygala24) [reply mteb/tasks/STS/multilingual/IndicCrosslingualSTS.py:70]: I have updated the dataset class to downsample the test set size of 256 per language with a stratified split. I don't see a significant drop in performance with a variance of 0.02 in correlation across a few languages.

File: mteb/tasks/STS/multilingual/IndicCrosslingualSTS.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                {"english_sentence": "sentence1", "indic_sentence": "sentence2"}
+            )
+            self.dataset[lang] = (
+                self.dataset[lang]
+                .map(lambda x: {"label": round(x["score"])})
+                .class_encode_column("label")
+            )
+            self.dataset[lang]["test"] = self.dataset[lang]["test"].train_test_split(
+                test_size=256,
+                seed=42,
```
Comment (KennethEnevoldsen) [inline mteb/tasks/STS/multilingual/IndicCrosslingualSTS.py:107]:
```suggestion
                seed=self.seed,
```

**Review by KennethEnevoldsen — APPROVED** [review]
Thanks for downsampling. Will merge this PR in
